packages/panna-sdk/src/react/components/activity/activity-list.tsx

```diff
@@ -105,12 +144,29 @@ export function ActivityList({ className }: ActivityListProps) {
     );
   }
 
+  const groupedActivities = groupActivitiesByDate(activitiesData);
+
   return (
     <section className="flex flex-col gap-6">
       {activitiesData.length > 0 ? (
-        activitiesData.map((activity) => (
-          <ActivityItem key={activity.transactionID} activity={activity} />
-        ))
+        Array.from(groupedActivities.entries()).map(([dateKey, activities]) => {
+          const date = new Date(dateKey);
+          return (
+            <div key={dateKey} className="flex flex-col gap-4">
+              <Typography variant="muted" className="text-sm">
+                {formatDateHeader(date)}
```

Comment:
We remove some unnecessary method calls by:
- Remove implementaiton above for `getDateKey`
- Rename `formatDateHeader` to `getDateKey`
- Make the following change (please verify for accuracy)

```suggestion
          return (
            <div key={dateKey} className="flex flex-col gap-4">
              <Typography variant="muted" className="text-sm">
                {dateKey}
```